File: SR-RBD/src/ddb/tpc/msg/TPCMessage.java
/**
 * 
 */
package ddb.tpc.msg;

import java.io.IOException;

import ddb.communication.DataInputStream;
import ddb.communication.DataOutputStream;
import ddb.msg.Message;

/** 
 * @author PKalanski
 */
public abstract class TPCMessage extends Message {
	/** 
	 */
	private String transactionId;

	/** 
	 * @return
	 */
	public String getTransactionId() {
		return transactionId;
	}

	/** 
	 * @param transactionId
	 */
	public void setTransactionId(String transactionId) {
		this.transactionId = transactionId;
	}
	
	@Override
	public void fromBinary(DataInputStream s) throws IOException {
		transactionId = s.readString();
		
	}

	@Override
	public void toBinary(DataOutputStream s) throws IOException {
		s.writeString(transactionId);
	}
}

File: trunk/C#/RBD/RBD/TPC/Msg/TPCMessage.cs
// +

using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

using RBD.Msg;
using RBD.Communication;

namespace RBD.TPC.Msg
{
    public abstract class TPCMessage : Message
    {

        public String TransactionId { get; set; }

        public String getTransactionId()
        {
            return TransactionId;
        }

        public void setTransactionId(String transactionId)
        {
            TransactionId = transactionId;
        }

        override
        public void FromBinary(DataInputStream s)
        {
            TransactionId = s.ReadString();
        }

        override
        public void ToBinary(DataOutputStream s)
        {
            s.WriteString(TransactionId);
        }
    }
}
